### PR TITLE
New sample for appliance Health

### PR DESCRIPTION
**Pull request by aspurohit**
Adding new sample to appliance .Sample demonstrates getting Health messages for various health items .Retrieves Health messages details from vCenter and prints the data.

### CODE REVIEW

File: samples/vsphere/appliances/health_messages.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        parser.add_argument(
+            '--item',
+            required=True,
+            action='store',
+            choices=['memory', 'cpu', 'storage'],
+            help='Specify the name of health item to view the messages')
+        args = sample_util.process_cli_args(parser.parse_args())
+        self.item = args.item
+
+    # Connect to vAPI services
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: FIx this indentation

File: samples/vsphere/appliances/health_messages.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            host=args.server,
+            user=args.username,
+            pwd=args.password,
+            skip_verification=args.skipverification)
+
+        self.health_client = Health(self.stub_config)
+
+    def run(self):
+        message_list = self.health_client.messages(self.item)
+        notification_message = Notification()
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: This is not required I guess

File: samples/vsphere/appliances/health_messages.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        msg = LocalizableMessage()
+        print(" Health Alarams")
+        print("-------------------\n")
+        if not message_list:
+            print("No health alarms for : " + self.item)
+        else:
+            for notification_message in message_list:
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: `for message in message_list`

File: samples/vsphere/appliances/health_messages.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                print("Alert time : {}".format(notification_message.time))
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: `message.time`

File: samples/vsphere/appliances/health_messages.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                print("Alert message Id: " + notification_message.id)
+                local_message = notification_message.message
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: `message.message`

File: samples/vsphere/appliances/health_messages.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                def_message = local_message.default_message
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: Make variable name def_message more descriptive

File: samples/vsphere/appliances/health_messages.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+from samples.vsphere.common import sample_util
+from samples.vsphere.common import vapiconnect
+from com.vmware.appliance_client import Health
+from com.vmware.vapi.std_client import LocalizableMessage
+from com.vmware.appliance_client import Notification
+
+
+class HealthMessages(object):
+    """
+    Demonstrates getting Health messages for various health items
```
[the remark this answers is not in the document]
Comment (pgbidkar) [reply samples/vsphere/appliances/health_messages.py]: `Demonstrates getting Health messages for memory, storage and cpu`

File: samples/vsphere/appliances/health_messages.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            host=args.server,
+            user=args.username,
+            pwd=args.password,
+            skip_verification=args.skipverification)
+
+        self.health_client = Health(self.stub_config)
+
+    def run(self):
+        message_list = self.health_client.messages(self.item)
+        message = Notification()
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: This is not required

File: samples/vsphere/appliances/health_messages.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        msg = LocalizableMessage()
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: this is also not required

File: samples/vsphere/appliances/health_messages.py [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+__author__ = 'VMware, Inc.'
+__copyright__ = 'Copyright 2019 VMware, Inc. All rights reserved.'
+__vcenter_version__ = '6.7+'
+
+from datetime import datetime, date
+
+from samples.vsphere.common import sample_cli
+from samples.vsphere.common import sample_util
+from samples.vsphere.common import vapiconnect
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: All three imports can be combined
```
from samples.vsphere.common import (sample_cli, sample_util, vapiconnect)

File: samples/vsphere/appliances/health_messages.py [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+    Retrieves Health messages details from vCenter and prints the data
+
+    """
+
+    def __init__(self):
+        self.item = None
+        self.stub_config = None
+
+    def setup(self):
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: You don't need setup. Just move setup code under __init__ to simplify the code.

File: samples/vsphere/appliances/health_messages.py [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+            required=True,
+            action='store',
+            choices=['memory', 'cpu', 'storage'],
+            help='Specify the name of health item to view the messages')
+
+        args = sample_util.process_cli_args(parser.parse_args())
+        self.item = args.item
+
+        # Connect to vAPI services
+        self.stub_config = vapiconnect.connect(
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Can you use vsphere_client instead? https://github.com/vmware/vsphere-automation-sdk-python/blob/master/sample_template/sample_template_basic.py#L52

File: samples/vsphere/appliances/health_messages.py [hunk 13]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Looks like the logic of the sample is not complex. Consider to create a basic sample using this format:
https://github.com/vmware/vsphere-automation-sdk-python/blob/master/sample_template/sample_template_basic.py

File: samples/vsphere/appliances/health_messages.py [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+*
+* DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+* EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+* WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+* NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+"""
+
+__author__ = 'VMware, Inc.'
+__copyright__ = 'Copyright 2019 VMware, Inc. All rights reserved.'
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Remove this duplicated copright message.

File: samples/vsphere/appliances/health_messages.py [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+        parser.add_argument(
+            '--item',
+            required=True,
+            action='store',
+            choices=['memory', 'cpu', 'storage'],
+            help='Specify the name of health item to view the messages')
+
+        args = sample_util.process_cli_args(parser.parse_args())
+        if args.item:
```
Comment (pgbidkar) [inline samples/vsphere/appliances/health_messages.py]: item argument is Mandatory parameter. I think this check is not required as arg parser it self will throw the error

File: samples/vsphere/appliances/health_messages.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (pgbidkar) [reply samples/vsphere/appliances/health_messages.py]: Please check if this change is taken care?

File: samples/vsphere/appliances/health_messages.py [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+class HealthMessages(object):
+    """
+    Demonstrates getting Health messages for memory, storage and cpu
+
+    Retrieves Health messages details from vCenter and prints the data
+
+    """
+
+    def __init__(self):
+        self.item = None
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: remove these two lines

File: samples/vsphere/appliances/health_messages.py [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+        args = sample_util.process_cli_args(parser.parse_args())
+        self.item = args.item
+        # Connect to vAPI services
+        session = get_unverified_session() if args.skipverification else None
+        self.client = create_vsphere_client(server=args.server,
+                                            username=args.username,
+                                            password=args.password,
+                                            session=session)
+        self.health_client = Health(self.client._stub_config)
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Remove this line.

File: samples/vsphere/appliances/health_messages.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+    def run(self):
+        message_list = self.health_client.messages(self.item)
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: replace health_client with self.client.appliance.Health

File: samples/vsphere/appliances/health_messages.py [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+* EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+* WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+* NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+"""
+
+__author__ = 'VMware, Inc.'
+__vcenter_version__ = '6.7+'
+
+from samples.vsphere.common import (sample_cli, sample_util)
+from com.vmware.appliance_client import Health
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Remove this line.

File: samples/vsphere/appliances/health_messages.py [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+from com.vmware.vapi.std_client import LocalizableMessage
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Looks like these two lines are not used.

File: samples/vsphere/appliances/health_messages.py [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+* WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+* EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+* WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+* NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+"""
+
+__author__ = 'VMware, Inc.'
+__vcenter_version__ = '6.7+'
+
+from samples.vsphere.common import (sample_cli, sample_util)
```
Comment (tianhao64) [inline samples/vsphere/appliances/health_messages.py]: Last comment:

Group imports from same package together and separate them with a line break. see "You should put a blank line between each group of imports." from https://www.python.org/dev/peps/pep-0008/#imports

from vmware.vapi.vsphere.client import create_vsphere_client

from samples.vsphere.common import (sample_cli, sample_util)
from samples.vsphere.common.ssl_helper import get_unverified_session

**Pull request merged**